crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -300,6 +309,92 @@ impl Schedule {
             system.apply_buffers(world);
         }
     }
+
+    /// Get the state of system-stepping
+    pub fn stepping(&self) -> bool {
+        match self.executable.step_state {
+            StepState::RunAll => false,
+            StepState::Wait(_) | StepState::Next(_) | StepState::Remaining(_) => true,
+        }
+    }
+
+    /// When stepping mode is enabled, this method will return the name of the
+    /// next system to be run when the schedule is stepped.
+    pub fn next_step_system_name(&self) -> Option<std::borrow::Cow<'static, str>> {
+        let index = self.executable.stepping_next_system_index()?;
+        Some(self.executable.systems[index].name())
+    }
+
+    /// When stepping mode is enabled, this method will return the [`NodeId`]
+    /// of the next system to be run when this schedule is stepped.
+    pub fn next_step_system_node_id(&self) -> Option<NodeId> {
+        let index = self.executable.stepping_next_system_index()?;
+        self.executable.system_ids.get(index).copied()
+    }
+
+    /// Check if a given system supports stepping
+    ///
+    /// This function will return false for an unknown `id`, or when the
+    /// schedule has not yet been built by calling [`Schedule::run()`].
+    pub fn system_permits_stepping(&self, id: NodeId) -> bool {
+        self.executable
+            .system_ids
+            .iter()
+            .position(|&i| i == id)
+            .map(|index| self.executable.systems_with_stepping_enabled[index])
+            .unwrap_or(false)
+    }
```

Comment:
Currently returning `false` here for unknown `NodeId`.  This function is used for stepping UI creation, and returns `false` in this case to simplify the logic there.

@alice-i-cecile would you prefer this be `Option(bool)` so the caller can detect & handle invalid `NodeId`?  I didn't have a need for it, but I do recognize we're losing some state in the return value.

Reply:
No strong feelings, but I do think `Option<Bool`> is a bit nicer.

Reply:
Due to the dependency here on `Schedule.executable` being built, this ended up being `Result<Option<bool>, ScheduleNotBuiltError`.  Same for all other helpers for accessing data from `Schedule.executable`.